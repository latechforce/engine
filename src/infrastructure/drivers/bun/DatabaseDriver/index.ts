import type { IDatabaseDriver } from '/adapter/spi/drivers/DatabaseSpi'
import type { DatabaseConfig, DatabaseEventType } from '/domain/services/Database'
import type { EventDto } from '/adapter/spi/dtos/EventDto'
import { PostgreSQLDatabaseDriver } from '/infrastructure/drivers/common/DatabaseDriver/PostgreSQLDriver'
import { SQLiteDatabaseDriver } from './SQLite/SQLiteDriver'
import type { AutomationHistoryRecord } from '/domain/entities/Automation/History'
import type { TableConfig } from '/domain/entities/Table'
import type { TableSchema } from '../../../../adapter/api/schemas/TableSchema'
import { TableMapper } from '/adapter/api/mappers/TableMapper'

export class DatabaseDriver implements IDatabaseDriver {
  private _db: PostgreSQLDatabaseDriver | SQLiteDatabaseDriver
  public type: DatabaseConfig['type']

  constructor(config: DatabaseConfig) {
    this.type = config.type
    switch (this.type) {
      case 'SQLite':
        this._db = new SQLiteDatabaseDriver(config)
        break
      case 'PostgreSQL':
        this._db = new PostgreSQLDatabaseDriver(config)
        break
      default:
        throw new Error('Invalid driver')
    }
  }

  connect = async (): Promise<void> => {
    await this._db.connect()
  }

  disconnect = async (): Promise<void> => {
    await this._db.disconnect()
  }

  exec = async (query: string): Promise<void> => {
    await this._db.exec(query)
  }

  query = async <T>(
    text: string,
    values: (string | number | Buffer | Date)[] = []
  ): Promise<{ rows: T[]; rowCount: number }> => {
    return this._db.query(text, values)
  }

  table(table: TableConfig) {
    return this._db.table(table)
  }

  tableFromSchema(schema: TableSchema) {
    return this._db.table(TableMapper.toConfig(schema))
  }

  on = (event: DatabaseEventType, callback: (eventDto: EventDto) => void) => {
    this._db.on(event, callback)
  }

  setupTriggers = async (tables: string[]) => {
    await this._db.setupTriggers(tables)
  }

  waitForAutomationsHistories = async ({
    count = 1,
    status = 'succeed',
    timeout = 60000,
  }: {
    count?: number
    status?: string
    timeout?: number
  } = {}) => {
    let histories: AutomationHistoryRecord[] = []
    let retry = 0
    do {
      await new Promise((resolve) => setTimeout(resolve, 500))
      const { rows } = await this.query<AutomationHistoryRecord>(
        'SELECT * FROM automations_histories_view WHERE status = ?',
        [status]
      )
      histories = rows
      retry++
    } while (histories.length < count && retry < Math.floor(timeout / 500))
    if (histories.length < count) {
      throw new Error(`Automations histories not found: ${count} ${status} histories`)
    }
    return histories
  }
}
