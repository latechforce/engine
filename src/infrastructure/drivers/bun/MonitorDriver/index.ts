import type { MonitorConfig } from '/domain/services/Monitor'
import type { IMonitorDriver } from '/adapter/spi/drivers/MonitorSpi'
import { SentryDriver } from './SentryDriver'
import { ConsoleDriver } from '../../common/MonitorDriver/ConsoleDriver'

export class MonitorDriver implements IMonitorDriver {
  private _monitors: (SentryDriver | ConsoleDriver)[] = []

  constructor(config: MonitorConfig[]) {
    for (const monitor of config) {
      const { driver } = monitor
      switch (driver) {
        case 'Sentry':
          this._monitors.push(new SentryDriver())
          break
        case 'Console':
          this._monitors.push(new ConsoleDriver())
          break
        default:
          throw new Error('Invalid driver')
      }
    }
  }

  captureException = (error: Error) => {
    this._monitors.map((m) => m.captureException(error))
  }

  captureMessage = (message: string) => {
    this._monitors.map((m) => m.captureMessage(message))
  }
}
