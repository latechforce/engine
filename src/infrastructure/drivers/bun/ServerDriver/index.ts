import type { IServerDriver } from '/adapter/spi/drivers/ServerSpi'
import type { DeleteDto, GetDto, PatchDto, PostDto, RequestDto } from '/adapter/spi/dtos/RequestDto'
import type { ServerConfig, ServerMethodOptions } from '/domain/services/Server'
import type { Response } from '/domain/entities/Response'
import { ElysiaDriver } from './ElysiaDriver'

export class ServerDriver implements IServerDriver {
  private _server: ElysiaDriver

  constructor(config: ServerConfig) {
    this._server = new ElysiaDriver(config)
  }

  get = async (
    path: string,
    handler: (getDto: GetDto) => Promise<Response>,
    options: ServerMethodOptions
  ) => {
    await this._server.get(path, handler, options)
  }

  post = async (
    path: string,
    handler: (postDto: PostDto) => Promise<Response>,
    options: ServerMethodOptions
  ) => {
    await this._server.post(path, handler, options)
  }

  patch = async (
    path: string,
    handler: (patchDto: PatchDto) => Promise<Response>,
    options: ServerMethodOptions
  ) => {
    await this._server.patch(path, handler, options)
  }

  delete = async (
    path: string,
    handler: (deleteDto: DeleteDto) => Promise<Response>,
    options: ServerMethodOptions
  ) => {
    await this._server.delete(path, handler, options)
  }

  notFound = async (handler: (requestDto: RequestDto) => Promise<Response>) => {
    await this._server.notFound(handler)
  }

  findAvailablePort = async (): Promise<number> => {
    return this._server.findAvailablePort()
  }

  start = async (port: number): Promise<number> => {
    return this._server.start(port)
  }

  stop = async () => {
    await this._server.stop()
  }
}
