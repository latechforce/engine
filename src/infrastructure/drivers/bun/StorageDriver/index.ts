import { PostgreSQLDriver } from '../../common/StorageDriver/PostgreSQLDriver'
import { SQLiteStorageDriver } from './SQLiteDriver'
import type { StorageConfig } from '/domain/services/Storage'
import type { IStorageDriver } from '/adapter/spi/drivers/StorageSpi'

export class StorageDriver implements IStorageDriver {
  private _storage: PostgreSQLDriver | SQLiteStorageDriver

  constructor({ driver, query, exec }: StorageConfig) {
    switch (driver) {
      case 'PostgreSQL':
        this._storage = new PostgreSQLDriver(query, exec)
        break
      case 'SQLite':
        this._storage = new SQLiteStorageDriver(query, exec)
        break
      default:
        throw new Error('Invalid driver')
    }
  }

  connect = async () => {
    await this._storage.connect()
  }

  disconnect = async () => {
    await this._storage.disconnect()
  }

  bucket = (name: string) => {
    return this._storage.bucket(name)
  }
}
