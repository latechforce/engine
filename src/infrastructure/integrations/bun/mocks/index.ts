import type { Integrations } from '/adapter/spi/integrations'
import type { NotionConfig } from '/domain/integrations/Notion'
import type { PappersConfig } from '/domain/integrations/Pappers'
import type { QontoConfig } from '/domain/integrations/Qonto'
import type { NgrokConfig } from '/domain/integrations/Ngrok'
import type { AirtableConfig } from '/domain/integrations/Airtable'
import type { GoogleMailConfig } from '/domain/integrations/Google/Mail'
import type { GoCardlessConfig } from '/domain/integrations/GoCardless'
import type { PhantombusterConfig } from '/domain/integrations/Phantombuster'
import type { CalendlyConfig } from '/domain/integrations/Calendly'
import type { YouCanBookMeConfig } from '/domain/integrations/YouCanBookMe'
import type { JotformConfig } from '/domain/integrations/Jotform/JotformConfig'
import type { ZoomConfig } from '/domain/integrations/Zoom/ZoomConfig'

import { NotionIntegration } from './notion/NotionIntegration.mock'
import { PappersIntegration } from './pappers/PappersIntegration.mock'
import { QontoIntegration } from './qonto/QontoIntegration.mock'
import { NgrokIntegration } from './ngrok/NgrokIntegration.mock'
import { AirtableIntegration } from './airtable/AirtableIntegration.mock'
import { GoogleMailIntegration } from './google/mail/GoogleMailIntegration.mock'
import { GoCardlessIntegration } from './gocardless/GoCardlessIntegration.mock'
import { PhantombusterIntegration } from './phantombuster/PhantombusterIntegration.mock'
import { CalendlyIntegration } from './calendly/CalendlyIntegration.mock'
import { YouCanBookMeIntegration } from './youcanbookme/YouCanBookMeIntegration.mock'
import { BaseMockIntegration } from './base'
import { JotformIntegration } from './jotform/JotformIntegration.mock'
import { ZoomIntegration } from './zoom/ZoomIntegration.mock'

export const mocks: Integrations = {
  airtable: (config: AirtableConfig) => new AirtableIntegration(config),
  notion: (config: NotionConfig) => new NotionIntegration(config),
  pappers: (config: PappersConfig) => new PappersIntegration(config),
  qonto: (config: QontoConfig) => new QontoIntegration(config),
  ngrok: (config: NgrokConfig) => new NgrokIntegration(config),
  googleMail: (config: GoogleMailConfig) => new GoogleMailIntegration(config),
  gocardless: (config: GoCardlessConfig) => new GoCardlessIntegration(config),
  phantombuster: (config: PhantombusterConfig) => new PhantombusterIntegration(config),
  calendly: (config: CalendlyConfig) => new CalendlyIntegration(config),
  youCanBookMe: (config: YouCanBookMeConfig) => new YouCanBookMeIntegration(config),
  jotform: (config: JotformConfig) => new JotformIntegration(config),
  zoom: (config: ZoomConfig) => new ZoomIntegration(config),
}

export { BaseMockIntegration }
