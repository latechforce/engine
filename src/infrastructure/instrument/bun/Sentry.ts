import type { MonitorSentryConfig } from '/domain/services/Monitor'
import * as Sentry from '@sentry/bun'
import slugify from 'slugify'

export function init(config: MonitorSentryConfig) {
  const release = `${slugify(config.appName, { lower: true })}@${config.appVersion}`
  Sentry.init({
    release,
    dsn: config.dsn,
    environment: config.environment,
    tracesSampleRate: 1.0,
  })
  console.log('Sentry has been successfully initialized')
}
