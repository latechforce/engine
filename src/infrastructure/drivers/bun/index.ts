import type { Drivers } from '/adapter/spi/drivers'
import type { DatabaseConfig } from '/domain/services/Database'
import type { MonitorsConfig } from '/domain/services/Monitor'
import type { ServerConfig } from '/domain/services/Server'
import type { StorageConfig } from '/domain/services/Storage'

import { DatabaseDriver } from './DatabaseDriver'
import { MonitorDriver } from './MonitorDriver'
import { ServerDriver } from './ServerDriver'

import { drivers as common } from '../common'
import { StorageDriver } from './StorageDriver'

export const drivers: Drivers = {
  ...common,
  monitor: (config: MonitorsConfig) => new MonitorDriver(config),
  database: (config: DatabaseConfig) => new DatabaseDriver(config),
  server: (config: ServerConfig) => new ServerDriver(config),
  storage: (config: StorageConfig) => new StorageDriver(config),
}
