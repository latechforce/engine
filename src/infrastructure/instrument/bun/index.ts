import type { ConfigSchema } from '/adapter/api/schemas/ConfigSchema'
import * as Sentry from './Sentry'
import { SystemDriver } from '/infrastructure/drivers/common/SystemDriver'

export function instrument(config: ConfigSchema) {
  const system = new SystemDriver()
  if (config.services?.monitors) {
    for (const monitor of config.services.monitors) {
      if (monitor.driver === 'Sentry')
        Sentry.init({
          appName: config.name,
          appVersion: config.appVersion ?? system.getAppVersion(),
          ...monitor,
        })
    }
  }
}
