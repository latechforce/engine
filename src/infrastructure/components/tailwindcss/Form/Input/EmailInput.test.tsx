import { it, expect } from 'bun:test'
import { EmailInput } from './EmailInput'
import { renderToString } from 'react-dom/server'

it('should match the email input snapshot', async () => {
  // GIVEN
  const input = <EmailInput field="email" />

  // WHEN
  const html = renderToString(input)

  // THEN
  expect(html).toMatchSnapshot()
})

it('should match the email input snapshot with label', async () => {
  // GIVEN
  const input = <EmailInput field="email" label="Email" />

  // WHEN
  const html = renderToString(input)

  // THEN
  expect(html).toMatchSnapshot()
})

it('should match the email input snapshot with description', async () => {
  // GIVEN
  const input = <EmailInput field="email" description="Description" />

  // WHEN
  const html = renderToString(input)

  // THEN
  expect(html).toMatchSnapshot()
})

it('should match the text input snapshot with label and description', async () => {
  // GIVEN
  const input = <EmailInput field="email" label="Email" description="Description" />

  // WHEN
  const html = renderToString(input)

  // THEN
  expect(html).toMatchSnapshot()
})

it('should match the email input snapshot with placeholder', async () => {
  // GIVEN
  const input = <EmailInput field="email" placeholder="Placeholder" />

  // WHEN
  const html = renderToString(input)

  // THEN
  expect(html).toMatchSnapshot()
})

it('should match the email input snapshot with required', async () => {
  // GIVEN
  const input = <EmailInput field="email" required />

  // WHEN
  const html = renderToString(input)

  // THEN
  expect(html).toMatchSnapshot()
})
