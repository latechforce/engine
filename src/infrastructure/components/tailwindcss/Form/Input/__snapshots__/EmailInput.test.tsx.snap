// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the email input snapshot 1`] = `"<div><input type="email" class="block w-full rounded-lg border border-gray-200 bg-white px-4 py-2.5 focus:border-blue-500 focus:ring-blue-500 disabled:pointer-events-none disabled:opacity-50 sm:py-3 sm:text-sm dark:border-neutral-700 dark:bg-neutral-900 dark:text-white dark:placeholder-neutral-500 dark:focus:ring-neutral-600" name="email"/></div>"`;

exports[`should match the email input snapshot with label 1`] = `"<div><label for="email" class="mb-2 block text-sm font-medium text-gray-700 dark:text-white">Email</label><input type="email" class="block w-full rounded-lg border border-gray-200 bg-white px-4 py-2.5 focus:border-blue-500 focus:ring-blue-500 disabled:pointer-events-none disabled:opacity-50 sm:py-3 sm:text-sm dark:border-neutral-700 dark:bg-neutral-900 dark:text-white dark:placeholder-neutral-500 dark:focus:ring-neutral-600" name="email"/></div>"`;

exports[`should match the email input snapshot with description 1`] = `"<div><p class="mb-2 text-sm text-gray-500 dark:text-neutral-500">Description</p><input type="email" class="block w-full rounded-lg border border-gray-200 bg-white px-4 py-2.5 focus:border-blue-500 focus:ring-blue-500 disabled:pointer-events-none disabled:opacity-50 sm:py-3 sm:text-sm dark:border-neutral-700 dark:bg-neutral-900 dark:text-white dark:placeholder-neutral-500 dark:focus:ring-neutral-600" name="email"/></div>"`;

exports[`should match the email input snapshot with label and description 1`] = `"<div><label for="email" class="mb-2 block text-sm font-medium text-gray-700 dark:text-white">Email</label><p class="mb-2 text-sm text-gray-500 dark:text-neutral-500">Description</p><input type="email" class="block w-full rounded-lg border border-gray-200 bg-white px-4 py-2.5 focus:border-blue-500 focus:ring-blue-500 disabled:pointer-events-none disabled:opacity-50 sm:py-3 sm:text-sm dark:border-neutral-700 dark:bg-neutral-900 dark:text-white dark:placeholder-neutral-500 dark:focus:ring-neutral-600" name="email"/></div>"`;

exports[`should match the email input snapshot with placeholder 1`] = `"<div><input type="email" placeholder="Placeholder" class="block w-full rounded-lg border border-gray-200 bg-white px-4 py-2.5 focus:border-blue-500 focus:ring-blue-500 disabled:pointer-events-none disabled:opacity-50 sm:py-3 sm:text-sm dark:border-neutral-700 dark:bg-neutral-900 dark:text-white dark:placeholder-neutral-500 dark:focus:ring-neutral-600" name="email"/></div>"`;

exports[`should match the email input snapshot with required 1`] = `"<div><input type="email" required="" class="block w-full rounded-lg border border-gray-200 bg-white px-4 py-2.5 focus:border-blue-500 focus:ring-blue-500 disabled:pointer-events-none disabled:opacity-50 sm:py-3 sm:text-sm dark:border-neutral-700 dark:bg-neutral-900 dark:text-white dark:placeholder-neutral-500 dark:focus:ring-neutral-600" name="email"/></div>"`;
